### PR TITLE
Clean up GetBindVolumeToClaim

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Refactored GetBindVolumeToClaim.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/volume/persistentvolume/util/util.go [hunk 1]
```diff
@@ -94,34 +94,24 @@ func IsDelayBindingMode(claim *v1.PersistentVolumeClaim, classLister storagelist
...
 	// modify these, therefore create a copy.
 	volumeClone := volume.DeepCopy()
 
-	// Bind the volume to the claim if it is not bound yet
-	if volume.Spec.ClaimRef == nil ||
-		volume.Spec.ClaimRef.Name != claim.Name ||
-		volume.Spec.ClaimRef.Namespace != claim.Namespace ||
-		volume.Spec.ClaimRef.UID != claim.UID {
-
+	// Check if the volume was already bound (either by user or by controller)
```
Comment (draveness) [inline pkg/controller/volume/persistentvolume/util/util.go]: Given the code of IsVolumeBoundToClaim, this refactor seems valid

```go
func IsVolumeBoundToClaim(volume *v1.PersistentVolume, claim *v1.PersistentVolumeClaim) bool {
	if volume.Spec.ClaimRef == nil {
		return false
	}
	if claim.Name != volume.Spec.ClaimRef.Name || claim.Namespace != volume.Spec.ClaimRef.Namespace {
		return false
	}
	if volume.Spec.ClaimRef.UID != "" && claim.UID != volume.Spec.ClaimRef.UID {
		return false
	}
	return true
}
```

**Review by draveness — APPROVED** [review]
/lgtm
/priority backlog
/ok-to-test

File: pkg/controller/volume/persistentvolume/util/util.go (hunk 1 above)
Comment (mrkm4ntr) [reply pkg/controller/volume/persistentvolume/util/util.go]: I didn't know that pre-bound volume and that it has empty `volume.Spec.ClaimRef.UID`. So changes of this PR became only removing an unnecessary intermediate variable and adding more comments about pre-bound volume.

**Comment by jsafrane** [comment]
I see it does not break anything, but is it worth removing one variable? Now the `if` looks slightly less readable to me.

**Comment by mrkm4ntr** [comment]
OK. I will close this. Thanks.

**Pull request closed without merging**